pkg/agent/openflow/packetin.go

```diff
@@ -99,29 +80,23 @@ func (c *client) StartPacketInHandler(packetInStartedReason []uint8, stopCh <-ch
 }
 
 func (c *client) subscribeFeaturePacketIn(featurePacketIn *featureStartPacketIn) error {
-	err := c.SubscribePacketIn(featurePacketIn.reason, featurePacketIn.subscribeCh)
+	err := c.SubscribePacketIn(featurePacketIn.reason, featurePacketIn.packetInQueue)
 	if err != nil {
-		return errors.New(fmt.Sprintf("Subscribe %d PacketIn failed %+v", featurePacketIn.reason, err))
+		return fmt.Errorf("subscribe %d PacketIn failed %+v", featurePacketIn.reason, err)
 	}
-	go c.parsePacketIn(featurePacketIn.packetInQueue, featurePacketIn.reason)
-	go featurePacketIn.ListenPacketIn()
+	go c.parsePacketIn(featurePacketIn)
 	return nil
 }
 
-func (c *client) parsePacketIn(packetInQueue workqueue.Interface, packetHandlerReason uint8) {
+func (c *client) parsePacketIn(featurePacketIn *featureStartPacketIn) {
 	for {
-		obj, quit := packetInQueue.Get()
-		if quit {
-			break
-		}
-		packetInQueue.Done(obj)
-		pktIn, ok := obj.(*ofctrl.PacketIn)
-		if !ok {
-			klog.Errorf("Invalid packetin data in queue, skipping.")
-			continue
+		pktIn := featurePacketIn.packetInQueue.GetRateLimited(featurePacketIn.stopCh)
+		if pktIn == nil {
+			klog.Infof("Received stop signal from stopCh.")
```

Comment:
I think the log message is not necessary

Reply:
Done.